adam/perception/perception_graph.py

```diff
@@ -1135,6 +1148,33 @@ def intersection(
         The algorithm used is approximate and is not guaranteed to return the largest
         possible match.
         """
+        attempted_match = self.intersection_getting_match(
+            graph_pattern,
+            debug_callback=debug_callback,
+            graph_logger=graph_logger,
+            ontology=ontology,
+            allowed_matches=allowed_matches,
+            match_mode=match_mode,
+            trim_after_match=trim_after_match,
+        )
+        if attempted_match:
+            return attempted_match.matched_pattern
+        else:
+            return None
```

Comment:
```suggestion
        return attempted_match.matched_pattern if attempted_match else None
```
Nit: This could be reduced to one line and be just as a clear.